### PR TITLE
Add Bedrock Agent example

### PR DESCRIPTION
**Pull request by moritalous**
#317

Add Bedrock Agent example

### CODE REVIEW

File: examples/bedrock-agent-fastapi/template.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  Function:
+    Timeout: 60
+
+Resources:
+  BedrockAgentFastAPIFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      PackageType: Image
+      MemorySize: 256
+      Policies: arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/template.yaml:19]: Here you can use policy name: 

```yaml
      Policies: AmazonS3ReadOnlyAccess
```

File: examples/bedrock-agent-fastapi/template.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    Metadata:
+      Dockerfile: Dockerfile
+      DockerContext: ./app
+      DockerTag: python3.12-v1
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/template.yaml:23]: You need to configure resource policy to allow Bedrock service to invoke this lambda function. And output the function arn will be useful. 

```yaml
  BedrockAgentPermission:
    Type: AWS::Lambda::Permission
    Properties:
      FunctionName: !Ref BedrockAgentFastAPIFunction
      Action: lambda:InvokeFunction
      Principal: bedrock.amazonaws.com
      SourceAccount: !Ref 'AWS::AccountId'
      SourceArn: !Sub arn:aws:bedrock:${AWS::Region}:${AWS::AccountId}:agent/*

Outputs:
  BedrockAgentFastAPIFunction:
    Description: "BedrockAgentFastAPIFunction Lambda Function ARN"
    Value: !GetAtt BedrockAgentFastAPIFunction.Arn
```

File: examples/bedrock-agent-fastapi/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+To deploy the application in your AWS account, you can use the SAM CLI's guided deployment process and follow the instructions on the screen
+
+```shell
+$ sam deploy --guided
+```
+
+Create a bedrock agent. (see [reference](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-create.html
+))
+
+## Test
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/README.md:51]: Before testing, we should show how to get the openapi spec from the FastAPI.

File: examples/bedrock-agent-fastapi/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Test your agent. (see [reference](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-test.html))
+
+## Run the docker locally
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/README.md:55]: This section could be removed and replaced with how to use `sam local invoke` to test locally the agent function with event payloads in the events directory.

**Review by bnusunny — CHANGES_REQUESTED** [review]

File: examples/bedrock-agent-fastapi/template.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+Globals:
+  Function:
+    Timeout: 60
+
+Resources:
+  BedrockAgentFastAPIFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      PackageType: Image
+      MemorySize: 256
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/template.yaml:18]: 256 is a bit slow. Increase it to 1024.

**Review by bnusunny — CHANGES_REQUESTED** [review]

File: examples/bedrock-agent-fastapi/app/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FROM public.ecr.aws/docker/library/python:3.12.0-slim
+COPY --from=public.ecr.aws/awsguru/aws-lambda-adapter:0.8.1 /lambda-adapter /opt/extensions/lambda-adapter
+ENV PORT=8000
```
Comment (bnusunny) [inline examples/bedrock-agent-fastapi/app/Dockerfile:3]: Add `AWS_LWA_READINESS_CHECK_PROTOCOL=tcp` to use TCP connect for healthcheck

```Dockerfile
ENV PORT=8000 AWS_LWA_READINESS_CHECK_PROTOCOL=tcp
```

**Review by bnusunny — CHANGES_REQUESTED** [review]

File: examples/bedrock-agent-fastapi/template.yaml (hunk 1 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/template.yaml]: fixed

File: examples/bedrock-agent-fastapi/template.yaml (hunk 2 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/template.yaml:23]: added your reccomend code

File: examples/bedrock-agent-fastapi/README.md (hunk 3 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/README.md]: add "Generate OpenAPI schema" section

File: examples/bedrock-agent-fastapi/README.md (hunk 4 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/README.md]: remove "Run the docker locally" section.
add "Test locally" section.

File: examples/bedrock-agent-fastapi/template.yaml (hunk 5 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/template.yaml]: fixed

File: examples/bedrock-agent-fastapi/app/Dockerfile (hunk 6 above)
Comment (moritalous) [reply examples/bedrock-agent-fastapi/app/Dockerfile]: fixed

**Review by bnusunny — APPROVED** [review]

**Comment by bnusunny** [comment]
Thanks for the contribution!

**Pull request merged**